bin/build_cmd.ml

```diff
@@ -193,7 +211,15 @@ let build =
       | _ :: _ -> targets
     in
     let common, config = Common.init builder in
-    let request setup =
+    let request (setup : Dune_rules.Main.build_system) =
+      let open Action_builder.O in
+      let* () =
+        Action_builder.of_memo
+        @@
+        if is_fmt_target ~common ~contexts:setup.contexts targets
```

Comment:
Build commands cannot have hacks like this. Maybe this logic should be moved to `dune fmt`.

Reply:
If understood what you mean:

I'm not opposed to move it to `dune fmt`. When users complains, we will figure out, how to support `dune build @fmt`.

Reply:
The main issue is that `dune build` is about building things. In other words, the scope of the work it can do is limited to setting up build rules and executing them. Adding functionality that isn't related to executing rules breaks the user's expectation in a fairly significant way. I suggested that `dune fmt` because that command does not have any expectations of being a regular build command. So we could add additional behavior to it.

Reply:
I removed this so now it only solves in `dune fmt` and not in `dune build @fmt`.